### ISSUE TITLE
Can't Lookup Existing UserGroup

### ISSUE THREAD

**Issue by Amorik** [Chaffelson/nipyapi#112]
* Nipyapi version: 0.12.0
* NiFi version: 1.7
* NiFi-Registry version: N/A
* Python version: 3.6
* Operating System: CentOS 7.x

### Description
I've successfully been able to modify the various policies in at the global and component level for a specific user. I am having trouble figuring out how to accomplish the same task for a group of users. The 'nipyapi.security.add_user_group_to_access_policy' function exists, but I do not see any way of obtaining/searching for a UserGroupEntity.

To be more specific my instance is setup with LDAP Authentication and when manually adding users I can select a group, which is oddly represented as a 'User'. When using 'nipyapi.security.list_service_users' groups do not show up as users, they only appear as components under user context in 'user_groups'. 

There is the 

### What I Did

#Gets root processor ID
root_pg_id = nipyapi.canvas.get_root_pg_id()
 #Gets user entity object if exists
user = nipyapi.security.get_service_user('myuser')
#Gets the current policy for the root processor group
policy = nipyapi.security.get_access_policy_for_resource('data', 'read', r_id=root_pg_id)
#Adds the user entity to the policy
nipyapi.security.add_user_to_access_policy(user, policy)

### Urgency

non-urgent.

**Comment by Amorik** [comment]
Ok so i dug in some more by building out simplified version of what 'get_service_user" essential does:
```
groups = getattr(nipyapi, 'nifi').TenantsApi().get_user_groups()
group = nipyapi.utils.filter_obj(groups.user_groups, 'My Group Name', 'identity')
```
This gave the error:
```
ValueError: UserGroupEntity is not a registered NiPyApi filterable class, registered classes are Bucket VersionedFlow RegistryClientEntity ProcessGroupEntity DocumentedTypeDTO ProcessorEntity User UserEntity TemplateEntity ControllerServiceEntity
```
The calls check against the result of ```nipyapi.config.registered_filters()```, which returns a simple dictionary of what appears to be a mapping of entity types and how to translate finding specific fields. This can bee seen when cross-referencing that with of ```TenantsApi().get_users()```and ```TenantsApi().get_user_groups```

To work around the issue I added the following mapping to registerd_filters dictionary:
```
addtofilter = {
	"UserGroup": {
		"identity": [
			"identity"
		],
		"id": [
			"identifier"
		]
	},
	"UserGroupEntity": {
		"identity": [
			"component",
			"identity"
		],
		"id": [
			"id"
		]
	},
}

nipyapi.config.registered_filters.update(addtofilter)
```

This effectively resolves the issue and returns the desired UserGroupEntity. Of course i do all that and then find out that in 0.12.0 the [nipyapi.security.add_user_group_to_access_policy()](https://nipyapi.readthedocs.io/en/latest/nipyapi-docs/nipyapi.html#nipyapi.security.add_user_group_to_access_policy) is not present. Is the removal of this function/method or absence of group lookup intentional?

**Comment by jrittenh** [comment]
`nipyapi.security.add_user_group_to_access_policy()` was [only recently merged to master](https://github.com/Chaffelson/nipyapi/commit/dac6601b06e1675e434fbe4428225717068f7b88) and has not been released yet.  You can install the latest unreleased code using pip: `pip install git+https://github.com/Chaffelson/nipyapi@master#egg=nipyapi` as a stopgap.

Also, [`nipyapi.security.get_access_policy_for_resource()` is buggy](https://github.com/Chaffelson/nipyapi/issues/110).  I haven't had time to dig into it further than the bug I filed, and @Chaffelson hasn't updated yet, either, so you may run into additional issues there.

**Comment by Chaffelson** [comment]
I'm going to try to pull working on this into the NiFi-1.9 update

**Comment by Chaffelson** [comment]
An update that this is in progress, I am now waiting on discussion of the Swagger bug affecting get_access_policy_for_resource before continuing

**Comment by Chaffelson** [comment]
This should now be fixed in the nifi191 branch

**Comment by Chaffelson** [comment]
Fixed in release 0.13, please reopen if there's still issues
